tests/units/units.cpp

```diff
@@ -2469,3 +2469,52 @@ TEST(Units, addUnitsMultipleTimes)
     // units.
     EXPECT_EQ(size_t(1), model->unitsCount());
 }
+
+TEST(Units, childUnitsWithIllegalPrefix)
+{
+    auto model = libcellml::Model::create("model");
+
+    auto u1 = libcellml::Units::create("u1");
+    auto u2 = libcellml::Units::create("u2");
+    auto u3 = libcellml::Units::create("u3");
+    auto u4 = libcellml::Units::create("u4");
+    auto u5 = libcellml::Units::create("u5");
+
+    model->addUnits(u1);
+    model->addUnits(u2);
+    model->addUnits(u3);
+    model->addUnits(u4);
+    model->addUnits(u5);
+
+    u5->addUnit("u4");
+    u4->addUnit("u3");
+    u3->addUnit("u2", "notAPrefix");
+    u2->addUnit("second");
+    u1->addUnit("second", "milli");
+
+    auto scaling = libcellml::Units::scalingFactor(u1, u5);
+    EXPECT_EQ(0.0, scaling);
+}
+
+TEST(Units, scalingFactorBetweenSameUnits)
+{
+    auto model = libcellml::Model::create("model");
+    auto u1 = libcellml::Units::create("u1");
+    u1->addUnit("second", "milli");
+    auto scaling = libcellml::Units::scalingFactor(u1, u1);
+    EXPECT_EQ(1.0, scaling);
+}
+
+TEST(Units, scalingFactorBetweenUnitsSameNameDifferentModels)
+{
+    auto model1 = libcellml::Model::create("model1");
+    auto u1 = libcellml::Units::create("units");
+    u1->addUnit("second", "milli");
+    model1->addUnits(u1);
+
+    auto model2 = model1->clone();
+    auto u2 = model2->units(0);
+
+    auto scaling = libcellml::Units::scalingFactor(u1, u2);
+    EXPECT_EQ(1.0, scaling);
+}
```

Comment:
Where is the co-test for this a non-unit scaling factor for units of different scale?

Reply:
There are loads of tests for scaling factors earlier in this file - I don't quite understand which aspect is missing?  This test was intended to show that even though these two have the same name, they're still tested properly.  I guess changing the  scale away from 1 would show that more clearly - is that what you meant?

Reply:
6008b53